javalib/src/main/scala/java/util/SplittableRandom.scala

```diff
@@ -0,0 +1,123 @@
+package java.util
+
+/*
+ * This is a clean room implementation derived from the original paper
+ * and Java implementation mentioned there:
+ *
+ * Fast Splittable Pseudorandom Number Generators
+ * by Guy L. Steele Jr., Doug Lea, Christine H. Flood
+ * http://gee.cs.oswego.edu/dl/papers/oopsla14.pdf
+ *
+ */
+private object SplittableRandom {
+
+  private final val DoubleULP = 1.0 / (1L << 53)
+  private final val GoldenGamma = 0x9e3779b97f4a7c15L // odd
+
+  private var defaultGen: Long = new Random().nextLong()
+
+  private def nextDefaultGen(): GenLong = {
+    val s = defaultGen
+    defaultGen = s + (2 * GoldenGamma)
+    GenLong(s)
+  }
+
+  case class GenLong(value: Long)
+
+  private final def mix64ForGamma(_z: Long): Long = {
+    var z = _z
+    z = (z ^ (z >>> 33)) * 0xff51afd7ed558ccdL
+    z = (z ^ (z >>> 33)) * 0xc4ceb9fe1a85ec53L
+    z ^ (z >>> 33)
+  }
```

Comment:
For (way) better performance, use `val`s with a different one at each line, i.e.,
```scala
  private final def mix64ForGamma(z: Long): Long = {
    val z1 = (z ^ (z >>> 33)) * 0xff51afd7ed558ccdL
    val z2 = (z1 ^ (z1 >>> 33)) * 0xc4ceb9fe1a85ec53L
    z2 ^ (z2 >>> 33)
  }
```
The reason is that, in Scala.js, `Long`s are reasonably fast only if the optimizer can crazy-optimize them. And the optimizer is not very good with `var`s; it is much better at dealing with `val`s.

The same comments apply to all 4 `mixXyz` methods (`mix64ForGamma`, `mix32`, `mix64` and `mixGamma`).

For `mix64ForGamma` only: it's also worth adding `@inline` on it, since it has only one call site. This will avoid boxing the parameter and result `Long`s.